source/common/common/utility.cc

```diff
@@ -98,12 +98,9 @@ void StringUtil::rtrim(std::string& source) {
   }
 }
 
-absl::string_view StringUtil::rightTrim(const absl::string_view& source) {
-  std::size_t pos = source.find_last_not_of(" \t\f\v\n\r");
-  if (pos != absl::string_view::npos) {
-    return absl::string_view(source.data(), pos + 1);
-  }
-  return absl::string_view("");
+absl::string_view StringUtil::rightTrim(absl::string_view source) {
```

Comment:
@jmarantz I've created a new class StringViewUtil with exactly the same method/implementation in my last commit. Check this out:  https://github.com/gsagula/envoy/blob/02ba40dd236645b89aec7689a81c9ff89f71dfc2/source/common/common/utility.cc#L222

Comment:
I think it's important to explicitly differentiate string view util from string util. Thoughts?

Reply:
In absl -- and in earlier versions of similar libraries in Google & Chromium, they are co-mingled.  Note that absl::string_view is I think intended as a stop-gap until C++17, when std::string_view (http://en.cppreference.com/w/cpp/string/basic_string_view) becomes available.

The reason -- I think -- that they are co-mingled is that there's no reason for the ones that take std::string arguments to exist at all.  E.g. in my example if you want to right-strip a std::string you can just pass in it, because absl::string_view has an implicit constructor from std::string, though it requires an explicit std::string out-conversion, e.g.:
    std::string foo;
    std::string bar = std::string(StringUtil::rightStrip(bar));
Ordinarily, though, if the backing-store stays alive along enough, you can just live in the string_view world and never make copies.

See https://github.com/abseil/abseil-cpp/blob/master/absl/strings/strip.h where the doc speaks in terms of std::string but actually all the functions take string_view.

So my preference is that you replace the ones in StringUtil:: rather than adding new alternatives.  This will help spread the knowledge about string_view's performance by eliminating the slower equivalent.

Having said that, I new your PR was coming, and the only reason I added my own version of rightStrip was as a testing vehicle for the benchmarking integration.  And I want the benchmarking for something unrelated (faster startup in the presence of giant configs).   Do you think you could split your StringUtil stuff in your PR into a smaller one, in which you also eliminate entirely the old rtrim and fix its call-sites?  Unfortunately the rtrim function's signature (modifying a std::string&) can't be fixed in place. 

What I'll do in any case is move my new utility function into the benchmark code itself, and your version can be authoritative, and I'll have a TODO on mine to delete it it and call yours instead once it's in.

Reply:
@jmarantz Thanks for the details.

Sounds good to me. I will open a small PR for StringUtil refactoring and some new methods that I need for #2087